app/main.py

```diff
@@ -1,2 +1,60 @@
+from typing import Any
+
+
+class Node:
+    def __init__(self, key: Any, value: Any) -> None:
+        self.key = key
+        self.value = value
+        self.hash = hash(key)
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.load_factor = 2 / 3
+        self.size = 0
+        self.table = [[] for _ in range(self.capacity)]
+
+    def __len__(self) -> int:
+        return self.size
+
+    def _hash(self, key: Any) -> int:
+        return hash(key) % self.capacity
+
+    def resize(self) -> None:
+        new_capacity = self.capacity * 2
+        new_table = [[] for _ in range(new_capacity)]
+
+        for bucket in self.table:
+            for node in bucket:
+                new_idx = hash(node.key) % new_capacity
```

Comment:
Consider using the _hash method instead of directly using hash(node.key) % new_capacity for consistency and to avoid potential errors if the hashing logic changes.

Reply:
Valid